src/DotVVM.Framework/ResourceManagement/InlineScriptResource.cs

```diff
@@ -22,23 +23,33 @@ public InlineScriptResource(string code, ResourceRenderPosition renderPosition =
         {
             this.Code = code;
         }
+        public InlineScriptResource(ILocalResourceLocation resourceLocation, ResourceRenderPosition renderPosition = ResourceRenderPosition.Body) : base(renderPosition)
+        {
+            this.resourceLocation = resourceLocation;
+        }
+
+        private ILocalResourceLocation resourceLocation;
+        private volatile Lazy<string> code;
 
         /// <summary>
         /// Gets or sets the javascript code that will be embedded in the page.
         /// </summary>
-        private string _code;
         public string Code
         {
-            get => _code;
+            get => code?.Value ?? throw new Exception("`ILocalResourceLocation` can not be read using property `Code`.");
             set
             {
                 InlineScriptContentGuard(value);
-                _code = value;
+                this.resourceLocation = new InlineResourceLocation(value);
+                this.code = new Lazy<string>(() => value);
             }
         }
 
         internal static void InlineScriptContentGuard(string code)
         {
+            // We have to make sure, that the element is not ended in the middle.
+            // <style> and <script> tags have "raw text" content - https://html.spec.whatwg.org/multipage/syntax.html#raw-text-elements
+            // and those element must not contain "</name-of-the-element" substring - https://html.spec.whatwg.org/multipage/syntax.html#cdata-rcdata-restrictions
             if (code?.IndexOf("</script", StringComparison.OrdinalIgnoreCase) >= 0)
                 throw new Exception($"Inline script can't contain `</script>`.");
```

Comment:
The error message is technically not true because the guard will prevent the user to use for example `</scriptContent>` or `</scripts>` which can be used in a string legitimately. I know it is difficult to restrict only `script` end tag because HTML allows attributes on closing tags, so never mind - we can just restrict that. But I would remove the `>` characted from the error message so it is more accurate.